test/chainrules.jl

```diff
@@ -1,17 +1,19 @@
 @testset "chainrules.jl" begin
-    @testset "BlockDiagonal" begin
-        x = [randn(1, 2), randn(2, 2)]
-        test_rrule(BlockDiagonal, x)
-    end
+    @testset for V in (Tuple, Vector)
+        @testset "BlockDiagonal" begin
+            x = V([randn(1, 2), randn(2, 2)])
+            test_rrule(BlockDiagonal, x)
+        end
 
-    @testset "Matrix" begin
-        D = BlockDiagonal([randn(1, 2), randn(2, 2)])
-        test_rrule(Matrix, D)
-    end
+        @testset "Matrix" begin
+            B = BlockDiagonal(V([randn(1, 2), randn(2, 2)]))
+            test_rrule(Matrix, B)
+        end
 
-    @testset "BlockDiagonal * Vector" begin
-        D = BlockDiagonal([rand(2, 3), rand(3, 3)])
-        v = rand(6)
-        test_rrule(*, D, v)
+        @testset "BlockDiagonal * Vector" begin
+            B = BlockDiagonal(V([rand(2, 3), rand(3, 3)]))
+            v = rand(6)
+            test_rrule(*, B, v)
```

Comment:
When the blocks are a tuple, this errors with
```julia
BlockDiagonal * Vector: Error During Test at /Users/npr/repos/BlockDiagonals.jl/test/chainrules.jl:13
  Got exception outside of a @test
  MethodError: no method matching +(::Composite{Tuple{Array{Float64,2},Array{Float64,2}},Tuple{Array{Float64,2},Array{Float64,2}}}, ::Array{InplaceableThunk{Thunk{BlockDiagonals.var"#11#17"{UnitRange{Int64},UnitRange{Int64},Array{Float64,1},Array{Float64,1}}},BlockDiagonals.var"#12#18"{UnitRange{Int64},UnitRange{Int64},Array{Float64,1},Array{Float64,1}}},1})
```

Reply:
changed so this PR just doesn't define any `rrule` methods for the `blocks::Tuple` case